method call on a binary object

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
    "crypto/ecdsa"
    "crypto/elliptic"
    "crypto/rand"
)   
    
func main() {
    key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
    if err != nil {
        panic(err)
    }
    pub := key.Public().(*ecdsa.PublicKey)
    println(pub.Params().Name)
}
```


### Expected result

```console
$ go run ./sample.go
P-256
```


### Got

```console
yaegi ./sample.go 
./sample.go:10:32: panic
run: reflect: Call with too few input arguments
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0x140002e6a00, 0x40)
	runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0x140001afbf0)
	github.com/traefik/yaegi/interp/interp.go:503 +0x94
panic(0x103029f00, 0x103161398)
	runtime/panic.go:965 +0x14c
github.com/traefik/yaegi/interp.runCfg.func1(0x140000ca8f0, 0x140000c5700, 0x140001af9c0)
	github.com/traefik/yaegi/interp/run.go:185 +0x1e8
panic(0x103029f00, 0x103161398)
	runtime/panic.go:965 +0x14c
reflect.Value.call(0x1030e0dc0, 0x14000010908, 0xe93, 0x102e38848, 0x4, 0x10364a870, 0x0, 0x0, 0x10314cb01, 0x1030e0dc0, ...)
	reflect/value.go:393 +0xe84
reflect.Value.Call(0x1030e0dc0, 0x14000010908, 0xe93, 0x10364a870, 0x0, 0x0, 0x140001af8c8, 0x10294a150, 0x140001af801)
	reflect/value.go:337 +0x84
github.com/traefik/yaegi/interp.callBin.func1(0x1030e0dc0, 0x14000010908, 0xe93, 0x10364a870, 0x0, 0x0, 0x0, 0x1030e0dc0, 0x140001af988)
	github.com/traefik/yaegi/interp/run.go:1330 +0x50
github.com/traefik/yaegi/interp.callBin.func10(0x140000ca8f0, 0x14000091b80)
	github.com/traefik/yaegi/interp/run.go:1499 +0x168
github.com/traefik/yaegi/interp.runCfg(0x140000c5700, 0x140000ca8f0)
	github.com/traefik/yaegi/interp/run.go:191 +0x8c
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000368000, 0x140000c4d00, 0x140000ca790)
	github.com/traefik/yaegi/interp/run.go:122 +0x244
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000368000, 0x1400000ab00, 0xfe, 0x16d8077de, 0x1a, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/traefik/yaegi/interp/interp.go:595 +0x5e8
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000368000, 0x16d8077de, 0x1a, 0x1400000aa00, 0xfe, 0x0, 0x0, 0x0)
	github.com/traefik/yaegi/interp/interp.go:417 +0xc4
main.runFile(0x14000368000, 0x16d8077de, 0x1a, 0x0, 0x0, 0x0)
	github.com/traefik/yaegi/cmd/yaegi/run.go:149 +0x84
main.run(0x14000032050, 0x1, 0x1, 0x14000001380, 0x1400006eea8)
	github.com/traefik/yaegi/cmd/yaegi/run.go:112 +0x830
main.main()
	github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x324
```


### Yaegi Version

v0.9.19

### Additional Notes

_No response_

--- a/_test/issue-1167.go
+++ b/_test/issue-1167.go
@@ -17,6 +17,3 @@ func main() {
 
 // Output:
 // P-256
-
-// Output:
-// P-256

--- a/_test/issue-1167.go
+++ b/_test/issue-1167.go
@@ -1,0 +1,22 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+)
+
+func main() {
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(err)
+	}
+	pub := key.Public().(*ecdsa.PublicKey)
+	println(pub.Params().Name)
+}
+
+// Output:
+// P-256
+
+// Output:
+// P-256

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1620,7 +1620,7 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 						n.val = method.Index
 						n.typ = &itype{cat: valueT, rtype: method.Type, recv: n.typ, isBinMethod: true}
 						n.recv = &receiver{node: n.child[0]}
-						n.gen = getIndexBinMethod
+						n.gen = getIndexBinElemMethod
 						n.action = aGetMethod
 					} else if method, ok := reflect.PtrTo(n.typ.val.rtype).MethodByName(n.child[1].ident); ok {
 						n.val = method.Index

--- a/interp/run.go
+++ b/interp/run.go
@@ -1510,6 +1510,20 @@ func getIndexBinMethod(n *node) {
 	}
 }
 
+func getIndexBinElemMethod(n *node) {
+	i := n.findex
+	l := n.level
+	m := n.val.(int)
+	value := genValue(n.child[0])
+	next := getExec(n.tnext)
+
+	n.exec = func(f *frame) bltn {
+		// Can not use .Set() because dest type contains the receiver and source not
+		getFrame(f, l).data[i] = value(f).Elem().Method(m)
+		return next
+	}
+}
+
 func getIndexBinPtrMethod(n *node) {
 	i := n.findex
 	l := n.level

